kedro/io/kedro_data_catalog.py

```diff
@@ -551,11 +551,82 @@ def add(
             )
         self.__setitem__(ds_name, dataset)
 
+    def filter(
+        self,
+        name_regex: str | None = None,
+        name_regex_flags: int | re.RegexFlag = re.IGNORECASE,
+        type_regex: str | None = None,
```

Comment:
When I've read the PR description about _filter by type_ I though that will be basically `instanceof` filtering, allowing to pass `Type` parameter, not string based class name, moreover a fully qualified one, which is more laborious to obtain for a plugin developer than just doing `data_catalog.filter(by_type=PluginSpecificDataset)`. WDYT?

Reply:
Oh I hadn't even considered that, I thought it would be the string representation people do in the YAML config! To be honest, both would be useful